firefox-ios/firefox-ios-tests/Tests/XCUITests/HomepageSearchBarTests.swift

```diff
@@ -0,0 +1,511 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+final class HomepageSearchBarTests: FeatureFlaggedTestBase {
+    func test_homepageSearchBar_experimentOff_doesNotShowSearchBar() {
+        addLaunchArgument(jsonFileName: "homepageSearchBarOff", featureName: "homepage-redesign-feature")
+        app.launch()
+        navigator.nowAt(NewTabScreen)
+        let homepageSearchBar = app.collectionViews
+            .cells.matching(identifier: AccessibilityIdentifiers.FirefoxHomepage.SearchBar.itemCell).element
+        let searchTextFieldA11y = AccessibilityIdentifiers.Browser.AddressToolbar.searchTextField
+
+        mozWaitForElementToNotExist(homepageSearchBar)
+        mozWaitForElementToExist(app.textFields[searchTextFieldA11y])
+    }
+
+    func test_homepageSearchBar_experimentOn_switchFromTopToBottomToolbar_showsExpectedBehavior() {
+        addLaunchArgument(jsonFileName: "homepageSearchBarOn", featureName: "homepage-redesign-feature")
+        app.launch()
+        navigator.nowAt(NewTabScreen)
+        navigator.goto(ToolbarSettings)
```

Comment:
Could you please confirm if the experiment does not apply to iPad? This test fails on iPad because the "Address Bar" is not found on settings.
<img src="https://github.com/user-attachments/assets/50568cf0-128f-4220-a9fb-54dfc8f5f936" width=200 />
You can put the snippet to ensure the test only runs on iPhone:
```
if !iPad() {
  // do stuff
}
```